### PR TITLE
Improve IP validation

### PR DESCRIPTION
**Pull request by raksooo**
This PR throws out the `ip` dependency (since it wasn't correct on which IPs were local and not) and replaces it with a custom implementation. The implementation contains parsing, validation and subnet matching of IPv4 and IPv6 addresses.

Git checklist:

* [ ] Describe the change in **`CHANGELOG.md`** under the `[Unreleased]` header.
* [x] Check that commits follow the [Mullvad coding guidelines](https://github.com/mullvad/coding-guidelines)

---
This change is [Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2461)

### CODE REVIEW

**Review by pronebird — APPROVED** [review]
:lgtm:

> Reviewed 5 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2461#-:-MTKaNR7CXXOj9eJO9b_:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved

File: gui/test/ip.spec.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  '0:1:2:3:4:5:6:7',
+  '00:11:22:33:44:55:66:77',
+  '000:111:222:333:444:555:666:777',
+  '0000:1111:2222:3333:4444:5555:6666:7777',
+  'ffff::',
+  '::ff:2233',
+  'fee::ff:2233',
+];
+
+const invalidIpv4Addresses = [
```
Comment (axti) [inline gui/test/ip.spec.ts:28]: Would it be good to have in test outside of range, something like 260.260.260.260 ?

Comment (raksooo) [reply gui/test/ip.spec.ts:28]: Thanks for the input! `10.0.0.256`, 256 is outside the range :)

**Review by pronebird — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2461)* status: all files reviewed, 1 unresolved discussion (waiting on @axti)

File: gui/test/ip.spec.ts (hunk 1 above)
Comment (raksooo) [reply gui/test/ip.spec.ts:28]: Please feel free to suggest more test cases. I haven't done a lot of IP validation/parsing/range comparisons before so I could definitely have made mistakes here.

**Review by pronebird — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2461)* status: all files reviewed, 1 unresolved discussion (waiting on @axti)

**Review by axti — APPROVED** [review]
:lgtm: 

> Reviewed 1 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2461#-:-MTKh9MGEJaJxT0iMd7S:bg7igkv)* status: 3 of 5 files reviewed, 1 unresolved discussion (waiting on @axti and @pronebird)

**Review by pronebird — COMMENTED** [review]
> Reviewed 1 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2461)* status: 4 of 5 files reviewed, all discussions resolved (waiting on @pronebird)

**Review by pronebird — APPROVED** [review]
> Reviewed 2 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/mullvad/mullvadvpn-app/2461)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**